scripts/switch-submodule-branch.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+git submodule foreach -q --recursive '
+  for branch in $(git config -f $toplevel/.gitmodules submodule.$name.branch); do
+    echo "Switching submodule $name to use branch $branch"
+    [ "$branch" = "" ] && git checkout main || git switch -q $branch
```

Comment:
You're assuming that every remote HEAD is `main` which isn't the case for all cardano repos. What if someone would like to use it with repo with different remote HEAD name, Like `cardano-base` ?

Reply:
For the `"$branch" = ""` case, it now generates a warning instead.  We can work from the assumption that if the branch isn't configured for a submodule then there's nothing to do.